### ISSUE TITLE
Sharepoint connector fetches documents only from the main site

### ISSUE THREAD

**Issue by akozichev** [airbytehq/airbyte#37003]
### Connector Name

source-microsoft-sharepoint

### Connector Version

0.2.2

### What step the error happened?

During the sync

### Relevant information

Sharepoint portals are structured in a way so they can host multiple site. Currently connector only recognises files hosted in the main site and everything else is ignored. The change is required so connector would iterate through all site when combining the files list.

### Contribute

- [X] Yes, I want to contribute

**Comment by isaac47** [comment]
Does the conenctor also return document's permission level? I am building a tool and I want to inherit from document permission level in other to manager user rights.

**Comment by nataliekwong** [comment]
Grooming notes:
- Most businesses create a different, custom site and don't use the default site. The default Sharepoint site is hard-coded as it assumes users use the default site, so this is a blocker.
- Allow users to specify a `site_id` or multiple `set_id` values in the source config. Leaving it empty would sync all discovered `site_id`

Example default site: https://{{company}}.sharepoint.com/Shared%20Documents/{{file name}}
Example custom site: https://{{company}}.sharepoint.com/sites/{{folder_name}}/Shared%20Documents/{{file name}}

fyi @natikgadzhi I added this to the community board, but let me know if the team is not working on file sources

**Comment by natikgadzhi** [comment]
Yeah, file sources are a bit out of scope for now, but the common bottleneck usually is design partners to verify test builds with. We'll get to this.

**Comment by natikgadzhi** [comment]
[We also potentially have design partners willing to test with us](https://airbytehq-team.slack.com/archives/C027KKE4BCZ/p1721169704021609).

**Comment by t-willi** [comment]
Has anything changed in regards to specifying the drive that airbyte should pull from?

**Comment by natikgadzhi** [comment]
@t-willi this got on our radar, but no, not yet.

**Comment by natikgadzhi** [comment]
@arr-ee another one, this one will be a bit more tricky.

**Comment by Mikkel-TS** [comment]
I am, amongst others, very hooked on implementing this update. If help is needed, I would be willing to invest time into testing/helping.
